fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationValidator.java

```diff
@@ -511,8 +517,10 @@ private void validateForCreate(final JsonElement element) {
 
             final LocalDate submittedOnDate = this.fromApiJsonHelper.extractLocalDateNamed(LoanApiConstants.submittedOnDateParameterName,
                     element);
-
             baseDataValidator.reset().parameter(LoanApiConstants.submittedOnDateParameterName).value(submittedOnDate).notNull();
+            if (submittedOnDate == null) {
```

Comment:
Same as above.

Reply:
The reason to add this return  there, is because previously we were generating a HTTP 400 Bad Request when some required field like expectedDisbursementDate or principal or transactionProcessingStrategy, etc are missed. 

Now with the new way, we have those schema validations and other business logic mixed, then the business validations are returning first than schema validations.